rules/RESPONSE-953-DATA-LEAKAGES-PHP.conf

```diff
@@ -103,8 +103,10 @@ SecRule RESPONSE_BODY "@rx <\?(?!xml)" \
     SecRule RESPONSE_BODY "!@rx (?:\x1f\x8b\x08|\b(?:(?:i(?:nterplay|hdr|d3)|m(?:ovi|thd)|r(?:ar!|iff)|(?:ex|jf)if|f(?:lv|ws)|varg|cws)\b|gif)|B(?:%pdf|\.ra)\b|^wOF[F2])" \
         "capture,\
         t:none,\
-        setvar:'tx.outbound_anomaly_score_pl1=+%{tx.error_anomaly_score}',\
-        setvar:'tx.anomaly_score_pl1=+%{tx.error_anomaly_score}'"
+        chain"
+        SecRule RESPONSE_HEADERS:Content-Encoding "!@streq br" \
+            "setvar:'tx.outbound_anomaly_score_pl1=+%{tx.error_anomaly_score}',\
+            setvar:'tx.anomaly_score_pl1=+%{tx.error_anomaly_score}'"
```

Comment:
Would it be more efficient to reverse the rules? String equlity check should be much less costly than worst case regexp evaluation.

Reply:
Ah, that's smart. I had not thought of that.
I will switch that. Thank you!

Reply:
Actually, I am not so sure of that. BC the current first rule is really selective. It will fail most of the time. With the reverse order, we run the non-selective rule first and will have to run the 2nd rule almost always on top. So from the perspective of REQUEST_BODY regex inspection not much changed, but an additional rule is running.

Reply:
Yes, but it's a "not" regexp... and the response text could be massive, I suspect? The regexp would be applied to each line in the response body... (I'm guessing here...)

Reply:
Let's think about this: the negation isn't part of the regular expression but of the annotation. This means that the regular expression will be applied until it finds a match or the end of the input is reached, only then can the result be checked and negated (unless the engine does some magic on top of PCRE). So for every line of input the expression engine will have to check every branch of the expression against the entire line to check for a match, by eye it looks like there are about 3 top level branches. Even with optimizations, I think the number of operations will be greater than for the string equality check (there's also some overhead for parsing the expression or looking it up from the cache).

Reply:
What should I do??

Reply:
Well, in reality it probably won't matter too much. The performance gain in either case would be marginal. You choose the explanation that sounds best to you, I suggest ;).